### ISSUE TITLE
FAHCore 0xa7 does not support Mac OS X 10.6 due to libc++ requirement

### ISSUE THREAD

**Issue by MCJack123** [FoldingAtHome/fah-issues#1315]
I tried installing Folding@Home on an old Mac that currently runs Mac OS X 10.6. I kept getting "FahCore returned: INTERRUPTED (102 = 0x66)". In the Console, I found that the core kept crashing:
```
Mon Mar  9 00:09:00 MacBook com.apple.ReportCrash.Root[1118] <Notice>: 2020-03-09 00:09:00.773 ReportCrash\[1118:2a03\] Saved crash report for FahCore_a7\[1116\] version ??? (???) to /Library/Logs/DiagnosticReports/FahCore_a7_2020-03-09-000900_localhost.crash
```
The crash report states that it was unable to find `libc++`, which is available on OS X 10.7 and later. 10.6 does not have `libc++` available whatsoever.

Here is the full crash report if needed:
```
Process:         FahCore_a7 [1116]
Path:            /Library/Application Support/FAHClient/cores/cores.foldingathome.org/v7/osx/64bit/Core_a7.fah/FahCore_a7
Identifier:      FahCore_a7
Version:         ??? (???)
Code Type:       X86-64 (Native)
Parent Process:  FAHCoreWrapper [1113]

Date/Time:       2020-03-09 00:09:00.757 -0400
OS Version:      Mac OS X 10.6.8 (10K549)
Report Version:  6

Exception Type:  EXC_BREAKPOINT (SIGTRAP)
Exception Codes: 0x0000000000000002, 0x0000000000000000
Crashed Thread:  0

Dyld Error Message:
  Library not loaded: /usr/lib/libc++.1.dylib
  Referenced from: /Library/Application Support/FAHClient/cores/cores.foldingathome.org/v7/osx/64bit/Core_a7.fah/FahCore_a7
  Reason: image not found

Binary Images:
    0x7fff5fc00000 -     0x7fff5fc3be0f  dyld 132.1 (???) <29DECB19-0193-2575-D838-CF743F0400B2> /usr/lib/dyld
```
Some possible solutions for this problem are:
1. Require OS X 10.7 or later to run Folding@Home. This may or may not be preferable depending on the concentration of older Macs in the current worker pool.
2. Make two different builds available: one for 10.7 and later, and one for 10.6 only. This would allow all systems currently advertised to work, but will increase the work required to build each version.
3. Ship a compatible version of `libc++` with Folding@Home. This guarantees that libc++ is available on all systems (but I'm not sure how easy this is considering `libc++` is a system library).

Here's an issue I found that discusses this issue as well: https://github.com/fish-shell/fish-shell/issues/3138

**Comment by MCJack123** [comment]
I was able to get Folding@Home working on my 10.6 machine after building libc++ myself. Here are the libraries that I used: [libc++-10.6.zip](https://github.com/FoldingAtHome/fah-issues/files/4310634/libc%2B%2B-10.6.zip). These should be copied into /usr/lib.

**Comment by bb30994** [comment]
Shouldn't the installer make this a prerequisite?

**Comment by MCJack123** [comment]
F@H officially requires 10.6 or later as stated on the download page:

<img width="550" alt="Screen Shot 2020-03-12 at 6 45 38 PM" src="https://user-images.githubusercontent.com/6236912/76572920-c9fc1d80-6491-11ea-8575-f88b8484e8ac.png">

And the installer checks for 10.6 or later in the Distribution file:
```xml
<volume-check script="volume_check();">
    <allowed-os-versions>
        <os-version min="10.6" />
    </allowed-os-versions>
</volume-check>
```

**Comment by bb30994** [comment]
So I think you're saying that FAH 7.5.1 actually add a requirement for 10.7 ... a fact which was not detected in beta testing?

**Comment by jcoffland** [comment]
The problem is that core 0xa7 for macOS requires 10.7+.  It might be possible to ship the libraries with the core.

**Comment by MCJack123** [comment]
I put together a patched installer package that installs `libc++` from the files above if on 10.6: https://drive.google.com/open?id=1C7TTEmCJXxt9hfAD9yAOmyLJhy7EKis0 . This adds a choice which is only enabled and available for 10.6.x that installs the required libraries. Note that this is untested, and the package ID for `libc++` uses my own identifiers instead of `org.foldingathome`.

@bb30994: I think this kind of issue could be easy to miss in testing because a) it's a bug that is somewhat hard to pinpoint (the UI doesn't show any issues unless you take a look at the logs), and b) it only appears on one version of an old operating system that may not be tested as well as other, more important systems.

**Comment by shorttack** [comment]
Closed: workaround by @MCJack123 shown above. Requesting MCJack123 try the latest beta at [https://foldingathome.org/beta/](url) to see if the next release is good on this issue. 

Note: Beta Page at https://foldingathome.org/beta/ still shows Mac 10.6 as minimum OS release. This may be a Documentation issue if 10.7 is the real oldest rev.
Thanks.
-Issues Team-
